src/client/retry.rs

```diff
@@ -33,7 +33,11 @@ use web_time::{Duration, Instant};
 
 /// Retry request error
 #[derive(Debug, thiserror::Error)]
-pub struct RetryError {
+pub struct RetryError(Box<RetryErrorImpl>);
```

Comment:
Isn't this a breaking public API change?

Reply:
I'm not sure RetryError is actually public, but regardless all the fields are crate private?

Reply:
I double checked and indeed I agree this error is not public
```shell
cargo doc --all-features --open
```

![Screenshot 2025-05-16 at 2 16 11 PM](https://github.com/user-attachments/assets/a73134e4-9b1d-42f2-8e51-0fc158a51ef5)